vulture_os/toolkit/api_parser/cybereason/cybereason.py

```diff
@@ -116,15 +105,23 @@ def _connect(self):
         except Exception as err:
             raise CybereasonAPIError(err)
 
-    def execute_query(self, method, url, query, timeout=10):
+    def execute_query(self, method, url, query=None, header=HEADERS, data=None, timeout=10):
         self._connect()
 
         response = None
         retry = 3
         while (retry > 0):
```

Comment:
```suggestion
        while (retry > 0 and not self.event_stop.is_set()):
```
prevent retries to keep the parser running for too long when asked to stop